app/artifact-cas/internal/server/grpc.go

```diff
@@ -130,39 +130,48 @@ func jwtAuthFunc(keyFunc jwt.Keyfunc, signingMethod jwt.SigningMethod) grpc_auth
 			return nil, err
 		}
 
-		var tokenInfo *jwt.Token
-		claims := &casJWT.Claims{}
-
-		tokenInfo, err = jwt.ParseWithClaims(token, claims, keyFunc)
+		claims, err := extractToken(token, keyFunc, signingMethod)
 		if err != nil {
-			var ve *jwt.ValidationError
-			if !errors.As(err, &ve) {
-				return nil, errors.Unauthorized("UNAUTHORIZED", err.Error())
-			}
-
-			if ve.Errors&jwt.ValidationErrorMalformed != 0 {
-				return nil, jwtMiddleware.ErrTokenInvalid
-			}
+			return nil, err
+		}
 
-			if ve.Errors&(jwt.ValidationErrorExpired) != 0 {
-				return nil, jwtMiddleware.ErrTokenExpired
-			}
+		return jwtMiddleware.NewContext(ctx, claims), nil
+	}
+}
 
-			if ve.Errors&(jwt.ValidationErrorNotValidYet) != 0 {
-				return nil, jwtMiddleware.ErrTokenExpired
-			}
+func extractToken(token string, keyFunc jwt.Keyfunc, signingMethod jwt.SigningMethod) (*casJWT.Claims, error) {
```

Comment:
I would rename this function to something like extractTokenAndVerify - you do way more here than a token extraction.